File: non_simple_py_action/src/simple_client.cpp
#include <non_simple_py_action/CountAction.h>
#include <actionlib/client/simple_action_client.h>
#include <err.h>

typedef actionlib::SimpleActionClient<non_simple_py_action::CountAction> Client;

int main(int argc, char** argv)
{
  ros::init(argc, argv, "simple_count_client");

  warnx("waiting for server");
  Client client("count", true); // true -> don't need ros::spin()
  client.waitForServer();

  warnx("sending goal");
  non_simple_py_action::CountGoal goal;
  goal.begin = -10;
  goal.end = 10;
  client.sendGoal(goal);
  
  warnx("waiting for result");
  client.waitForResult();
  if (client.getState() == actionlib::SimpleClientGoalState::SUCCEEDED)
    printf("Yay!\n");
  printf("Current State: %s\n", client.getState().toString().c_str());
  return 0;
}

File: non_simple_action/scripts/simple_client.py
#! /usr/bin/env python

import sys
import os
import roslib
roslib.load_manifest('non_simple_action')

import rospy
import actionlib
import non_simple_action.msg

if __name__ == '__main__':
    rospy.init_node('count_client')
    sac = actionlib.SimpleActionClient('count',
                                       non_simple_action.msg.CountAction)
    print 'waiting for server'
    sac.wait_for_server()
    print 'sending goal'
    goal = non_simple_action.msg.CountGoal(begin = -10, end = 10)
    sac.send_goal(goal)
    print 'waiting for result'
    sac.wait_for_result()
    result = sac.get_result()
    print 'result: %s' % result
